src/charm.py

```diff
@@ -58,17 +58,26 @@ def _on_update_status(self, _) -> None:
     def _on_config_changed(self, _) -> None:
         """Handle changes in configuration."""
         self.unit.status = ops.MaintenanceStatus("Configuring charm")
+        self._store_password_as_secret()
         self._update_relations()
         self.unit.status = ops.ActiveStatus()
 
+    def _store_password_as_secret(self) -> None:
+        """Store the SMTP password as a secret."""
+        if self._charm_state.password:
+            secret = self.app.add_secret({"password": self._charm_state.password})
```

Comment:
For every config-changed event, a new secret will be created?

Reply:
BTW where happens the get_secret?

Reply:
You have a suggestion, right?

Reply:
Well, I'm not sure if this is the best approach for the SMTP as well but here in Synapse we decided for doing similar to Indico by checking if Secrets is supported, storing the ID in peer data and then, when needed, this ID is used to get the Secret. Also, while getting the Secret, if for some reason it doesn't exist, the peer data is erased and an error is raised.

get_admin_access_token in https://github.com/canonical/synapse-operator/pull/59/files#diff-b9ed39bbc9c0387bd3e07da31d13373745534a1cd723d3e292c73496b12e307c

Reply:
I'll be checking if secrets are supported in a follow up PR. It's stated in the description ;)